### PR TITLE
Badstatus

### PR DESCRIPTION
**Pull request by zerc**
In case of incoming major release:

* Transform the `ucare_cli.py` into submodule coz, it was so big.
* Added ability to continue syncing from point where an error happened.
* Added ability to retry of unsuccessful downloads.

[Review on Reviewable](https://reviewable.io/reviews/uploadcare/pyuploadcare/105)

### CODE REVIEW

**Comment by dmitry-mukhin** [comment]
Reviewed 1 of 2 files at r1, 5 of 5 files at r2, 3 of 3 files at r3, 2 of 2 files at r4, 2 of 2 files at r5, 2 of 2 files at r6, 2 of 2 files at r7.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/uploadcare/pyuploadcare/105)*

**Comment by dmitry-mukhin** [comment]
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[a discussion](https://reviewable.io:443/reviews/uploadcare/pyuploadcare/105#-KBwFtWAs8064VwVLrZ1:-KBwFtWAs8064VwVLrZ2:-1909544718) (no related file):*
LGTM, @homm, your turn.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/uploadcare/pyuploadcare/105)*

**Comment by dmitry-mukhin** [comment]
LGTM

---

Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/uploadcare/pyuploadcare/105#-:-KBwGSJ2rQezhnxJ5cDu:676834233)*

**Comment by dmitry-mukhin** [comment]
wtf :)

**Comment by dmitry-mukhin** [comment]
one more time: :lgtm:

---

Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/uploadcare/pyuploadcare/105#-:-KBwGlJcgmjJ1-ICv-YW:64935371)*

**Comment by homm** [comment]
А где сам фикс? Очень сложно понять, когда столько перемещений кода.

---

Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/uploadcare/pyuploadcare/105#-:-KBwuMU04H_lpBHdBSGw:-1151538926)*

File: pyuploadcare/ucare_cli/utils.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+def int_or_none(value):
+    return None if value.lower() == 'none' else int(value)
+
+
+def promt(text, default='y'):
+    return (input('{0} [y/n]: '.format(text)) or default) == 'y'
+
+
+def bar(iter_content, parts, title=''):
+    """ Iterates over the "iter_content" and draws a progress bar to stdout.
```
Comment (homm) [inline pyuploadcare/ucare_cli/utils.py]: https://pypi.python.org/pypi/tqdm ?

**Comment by zerc** [comment]
*[pyuploadcare/ucare_cli/utils.py, line 27 \[r7\]](https://reviewable.io:443/reviews/uploadcare/pyuploadcare/105#-KBwBPcyJp7q2dDfAYd7-r7-27:-KC0lYhjSLscdIrsxoTN:2045219005) ([raw file](https://github.com/uploadcare/pyuploadcare/blob/375c1e40bbc08a2263ea8d5f56571c7fa48a7496/pyuploadcare/ucare_cli/utils.py#L27)):*
Развесистая штука. Но если ты не против еще одной депенденси - то я только за.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/uploadcare/pyuploadcare/105)*

**Comment by homm** [comment]
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[pyuploadcare/ucare_cli/utils.py, line 27 \[r7\]](https://reviewable.io:443/reviews/uploadcare/pyuploadcare/105#-KBwBPcyJp7q2dDfAYd7-r7-27:-KC0ls1Krm1Qt0Xi8M6m:348880587) ([raw file](https://github.com/uploadcare/pyuploadcare/blob/375c1e40bbc08a2263ea8d5f56571c7fa48a7496/pyuploadcare/ucare_cli/utils.py#L27)):*
Ну дискуссионный вопрос. Если ткущей bar хватате и она не глючит, можно оставить как есть.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/uploadcare/pyuploadcare/105)*

**Comment by zerc** [comment]
https://github.com/uploadcare/pyuploadcare/pull/105/files#diff-210174c0a9939fe0e41fe7d13e7f8c1dR26 - ретрай запроса.

https://github.com/uploadcare/pyuploadcare/pull/105/files#diff-210174c0a9939fe0e41fe7d13e7f8c1dR129 - отсюда и ниже идет реализация восстановления состояния

---







*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/uploadcare/pyuploadcare/105#-:-KC0lgv3cR1L4egyYZOj:2106471044)*

**Comment by zerc** [comment]
*[pyuploadcare/ucare_cli/utils.py, line 27 \[r7\]](https://reviewable.io:443/reviews/uploadcare/pyuploadcare/105#-KBwBPcyJp7q2dDfAYd7-r7-27:-KC0m8TreuUoarW6kTur:515448603) ([raw file](https://github.com/uploadcare/pyuploadcare/blob/375c1e40bbc08a2263ea8d5f56571c7fa48a7496/pyuploadcare/ucare_cli/utils.py#L27)):*
Текущий bar уже давно тут сидит, вроде все впорядке. 

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/uploadcare/pyuploadcare/105)*

**Comment by homm** [comment]
:lgtm:

---

Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/uploadcare/pyuploadcare/105#-:-KDOq1IS0Ip2A9oWUttQ:1417088721)*

**Pull request merged**